src/controllers/products.controller.ts

```diff
@@ -108,6 +108,41 @@ const getLimitedEdtionProducts: RequestHandler<
   res.json({ count, rows });
 };
 
+const searchProducts: RequestHandler<
+  object,
+  object,
+  object,
+  PaginationQuery & { keyword: string }
+> = async (
+  req: Request<object, object, object, PaginationQuery & { keyword: string }>,
+  res: Response
+) => {
+  const page = req.query.page ? parseInt(req.query.page) : 1;
+  const perPage = req.query.perPage ? parseInt(req.query.perPage) : 1;
+  const keyword = req.query.keyword ?? '';
+
+  const { count, rows } = await Product.findAndCountAll({
+    where: {
+      [Op.or]: [
+        { name: { [Op.like]: `%${keyword}%` } },
+        { '$brand.name$': { [Op.like]: `%${keyword}%` } }
+      ]
+    },
+    include: [
+      {
+        model: Brand,
+        where: {}
```

Comment:
No need for the empty `where`

Reply:
Yes, I believe there is an issue with _sequelize_, If I didn't put the where, it won't recognize the attribute _name_ of the Brand table.